internal/charmstore/store_test.go

```diff
@@ -128,6 +128,36 @@ func (s *StoreSuite) checkAddCharm(c *gc.C, ch charm.Charm, addToES bool, url *r
 	c.Assert(errgo.Cause(err), gc.Equals, params.ErrDuplicateUpload)
 }
 
+// addRequiredCharms adds any charms required by the given
+// bundle that are not already in the store.
+func (s *StoreSuite) addRequiredCharms(c *gc.C, bundle charm.Bundle) {
```

Comment:
This seems to be implemented three times, for charmstore, v4 and v5.
Should we define a storetesting.AddRequiredCharms?